test/src/utils/kalman_multivariate.test.cpp

```diff
@@ -104,6 +104,74 @@ TEST_F(KalmanFunctionality, handlesArbitraryStateCovariance)
   ASSERT_EQ(kalman.getStateCovariance(), P) << arb_covariance_err;
 }
 
+/**
+ * Test fixture used for determining whether Kalman handles an update on the State transition matrix
+ * It will be updated twice, first to the Zero Matrix and then to to Identity Matrix
+ */
+TEST_F(KalmanFunctionality, handlesUpdateInA)
+{
+  KalmanMultivariate kalman_two = KalmanMultivariate(n, m, 0);
+  VectorXf x1 = VectorXf::Random(n);
+  while (x1 == VectorXf::Zero(n)) {
+    x1 = VectorXf::Random(n, n);
+  }
+  A = MatrixXf::Random(n, n);
+  while (A == MatrixXf::Zero(n, n) || A == MatrixXf::Identity(n, n)) {
+    A = MatrixXf::Random(n, n);
+  }
+  B = MatrixXf::Zero(n, k);
```

Comment:
Just a quick question, do the variable names have a meaning i.e. a specific matrix? If they do, is it best to add comments next to each assignment to let others know which matrix is which?

Reply:
I think they are standard but they might have names I will research on that

Reply:
For example, x is the state vector and P is the Covariance

Reply:
Awesome ok! Think some sort of commenting or documentation would be great!